fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/MutableRecordStoreState.java

```diff
@@ -145,10 +154,22 @@ public IndexState setState(@Nonnull String indexName, @Nonnull IndexState state)
         return previous == null ? IndexState.READABLE : previous;
     }
 
+    /**
+     * Modify the meta-data of an index in this map. The caller should modify the database to match.
+     *
+     * @param indexName the index name
+     * @param indexMetaData the new state meta-data for the given index
+     */
+    @Nonnull
+    public void setIndexMetaData(@Nonnull String indexName, @Nonnull IndexMetaDataProto.IndexMetaData indexMetaData) {
+        verifyWritable();
+        indexMetaDataMap.get().put(indexName, indexMetaData);
```

Comment:
In the same vein as my comment above, if we keep things written this way, maybe this should verify that the metadata being set is for the current state.  Or, probably better, require that the state and the metadata be set together?